doc/Makefile

```diff
@@ -19,25 +19,68 @@ default: container-build
 # should not be used with `make -f`.
 ##
 
+CONTAINER_VERSION=1.1.1
+CONTAINER_NAME=openassetio-doc-build
+
+##
+## Container settings
+##
+
+# The following variables are considered when using Docker
+# to build the documentation:
+#
+#  GITHUB_ORG: The organisation to pull containers from (and publish
+#              to). If unspecified, then `thefoundryvisionmongers` will
+#              be used. Note: org names must be lowercase.
+#
+# When publishing, the following additional vars MUST be set:
+#
+#  GITHUB_PAT:      A GitHub Personal Access Token with the following
+#                   permissions: write:packages.
+#  GITHUB_USERNAME: The GitHub username whose token is being used.
+#
+# eg:
+#  GITHUB_USER=<username> GITHUB_PAT=<token> make publish-docker
+#
+ifndef GITHUB_ORG
+GITHUB_ORG = thefoundryvisionmongers
+endif
+
+# Docker images are published to the GitHub container registry
+CONTAINER_PATH = ghcr.io/$(GITHUB_ORG)/$(CONTAINER_NAME)
+CONTAINER_TAG = $(CONTAINER_PATH):$(CONTAINER_VERSION)
 
 ##
 ## PHONY TARGETS
 ##
 
-.PHONY: clean clean-html clean-tooling clean-docker container-build deploy docker-image html todo tooling
+.PHONY: clean clean-html clean-tooling clean-docker container-build deploy docker-image html \
+        publish-docker tag-docker todo tooling
 
 #
 # Container build - Runs this Makefile via docker
 #
 
-container-build: docker-image
-	docker run --rm -v `pwd`/../:/src openassetio-doc-build bash -c 'make -C /src/doc html'
+container-build:
+	docker run --rm -v `pwd`/../:/src $(CONTAINER_TAG) bash -c 'make -C /src/doc html'
+
+# Targets for building/tagging/publishing the container
 
 docker-image: Dockerfile
-	docker build . -t openassetio-doc-build
+	docker build . -t $(CONTAINER_NAME)
 
 clean-docker:
-	docker image rm openassetio-doc-build --force
+	docker image rm $(CONTAINER_NAME) --force
```

Comment:
[consider] This removes the `$CONTAINER_NAME:latest` image, but not the `$CONTAINER_TAG` image.

It looks like there is (still) no built-in way to delete all images for a given "repository" (`CONTAINER_PATH`). Simplest solution is to delete the current `CONTAINER_TAG` image and assume the tag hasn't been bumped since the image was last built. Otherwise shell grepping is required.

Reply:
I checked and deleting the "parent" `CONTAINER_NAME` image then rebuilding is almost a no-op, since docker just re-tags the `CONTAINER_TAG` image as `CONTAINER_NAME` - the image ID is identical.

Reply:
This is better now with the above change to always using `CONTAINER_TAG`. You can use `clean-docker` to remove your local build of the current image version - which could be enough for now?